apinf/api-umbrella-dashboard
# Feature/517 API calls thorough umbrella proxy - closes #517

@apinf/developers Please review. #517 related.

#### The approach is
1. Read the text file when selected.
2. Convert text data to JSON object to be able to refer to its `values` by `key`.
3. Modify values of `host` and `basePath` to what we need.
4. Convert object back to JSON string.
5. Construct `new File( )` using updated file data. (This is implemented in two steps: First constructing `new Blob( )`. Second: `new File( )` that takes `Blob` as parameter)
6. Saving file in Mongodb collection

#### Issues
* Currently custom values of `basePath` and `host` are hardcoded. Since all the above process is happening on a client-side, there is no way to access Meteor.settings data.

#### Testing 
1. Create new `apiBackend`. While this process, upload new documentation file e.g swagger.json. (For example default one: http://petstore.swagger.io/v2/swagger.json)
2. Go to documentation viewer page.
3. In drop down manu select uploaded file.
4. iFrame will give an error since the remote swagger UI was not able to access local file. 
5. The path to documentation file is provided in there. 
6. Copy this path to a new tab to see updated `host` and `basePath` values.
